common/src/main/java/org/apache/falcon/entity/FeedHelper.java

```diff
@@ -437,10 +436,20 @@ public static Date getNextFeedInstanceDate(Date alignedDate, Feed feed) {
         Cluster cluster = getCluster(feed, clusterName);
         if (cluster != null) {
             if (isLifecycleEnabled(feed, clusterName)) {
-                String policy = getRetentionStage(feed, clusterName).getPolicy();
-                policy = StringUtils.isBlank(policy)
-                        ? FeedLifecycleStage.RETENTION.getDefaultPolicyName() : policy;
-                result.add(policy);
+                String policy = "";
```

Comment:
getPolicies method : rather than return the generic list of all policies across stages, please have separate methods. Clubbing policies across lifecycle stages does not make sense.